apps/prairielearn/python/zygote_utils_test.py

```diff
@@ -1,9 +1,36 @@
+import json
 from typing import Any
 
 import pytest
 import zygote_utils as zu
 
 
+@pytest.mark.parametrize(
+    "item", ["-9007199254740991", "-1", "0", "1", "9007199254740991"]
+)
+def test_safe_parse_int_small_ints(item: str) -> None:
+    loaded_item = json.loads(item, parse_int=zu.safe_parse_int)
+    assert isinstance(loaded_item, int)
+    assert int(item) == loaded_item
+
+
+@pytest.mark.parametrize(
+    "item",
+    [
+        "-2.8e16",
+        "28000000000000000",
+        "-9007199254740992",
+        "9007199254740992",
+        "28000000000000000",
+        "2.8e16",
+    ],
+)
+def test_safe_parse_int_large_ints(item: str) -> None:
+    loaded_item = json.loads(item, parse_int=zu.safe_parse_int)
+    assert isinstance(loaded_item, float)
+    assert float(item) == loaded_item
```

Comment:
One thing to note: You should basically never compare floats in Python with `==`. In this case however, it's actually better for testing since we want to assert that the string `item` itself was converted to a float using `float(item)`.

Reply:
Interesting! What's the recommended way to do that in normal scenarios?

Reply:
The standard Python-library way to do it is with `math.isclose()` (comparison relative to small fixed tolerances), and with Numpy objects `np.isclose()` is commonly used. There's a detailed discussion of these issues here: https://saturncloud.io/blog/what-is-float-comparison-in-pandas-and-how-to-do-it/

Reply:
to be clear, this is floats in any language, not just python